### PR TITLE
feat(cardeditor): initial drop of experimental component

### PR DESCRIPTION
**Pull request by bryancboyd**
**Summary**

This PR provides an initial implementation that covers work for the following issues:

#1560 
#1561 
#1562 
#1563 

We'll provide further improvements and additions to this component over the next 2 weeks, but the component delivered in this PR is functional enough to start being imported and used by applications

**Change List (commits, features, bugs, etc)**

- Added new Experimental component `<CardEditor>`

**Acceptance Test (how to verify the PR)**

- Jest tests pass
- Click around the storybook `<CardEditor>`interactive story.  The JSON editor allows you to fully modify the JSON definition of the card, and the card edit form will provide inputs to modify card size and title.  If no card definition is passed to CardEditor, the component opens in "gallery mode" where the user can select a card type from a list.

The interactive story also renders the card definition JSON on the left side of the screen, updated in response to the `onChange` event defined on CardEditor.

image

### CODE REVIEW

File: src/components/CardEditor/CardEditForm/_card-edit-form.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    padding-bottom: $spacing-05;
+  }
+  &--footer {
+    & > button {
+      width: 100%;
+    }
+  }
+  .temp-json-editor {
+    textarea {
+      height: 30rem;
```
Comment (sstone2423) [inline src/components/CardEditor/CardEditForm/_card-edit-form.scss:21]: This can be added to the actual TextArea component with the `columns` prop

**Review by sstone2423 — CHANGES_REQUESTED** [review]
Looks good, but is there any kind of state to go back to the existing card that is being editted? If you start in the default story, then click `view gallery`, theres no way of going back to the existing card that is being editted

**Comment by bryancboyd** [comment]
Yeah, in fact I was thinking about removing the showGallery state from the component completely and replacing with a callback (onShowGallery) so that the implementer can manage the "mode" of the editor component (add (gallery) or edit (form)). I'll do that for now.

In proposed "Dashboard Editor" patterns I've seen, the Card Editor would default to the "gallery" state when no cards are selected on the preview dashboard. Then, when either A) a user selects an existing card, or B) a user selects a card type from the gallery, the editor would switch over to the "form" state. With this behavior, selecting "View gallery" would essentially deselect the currently-editing card.

(I wouldn't be surprised if the "view gallery" button gets removed by the time we are finished)

**Review by sstone2423 — APPROVED** [review]

**Review by filipecorrea — DISMISSED** [review]
`Table` and `TableViewDropdown` snapshots needs to be updated.

File: src/components/CardEditor/CardEditor.test.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      <CardEditor
+        value={defaultCard}
+        onShowGallery={actions.onShowGallery}
+        onChange={actions.onChange}
+        onAddCard={actions.onAddCard}
+      />
+    );
+    userEvent.click(
+      screen.getByRole('button', {
+        name: CardEditor.defaultProps.i18n.openGalleryButton,
```
Comment (tay1orjones) [inline src/components/CardEditor/CardEditor.test.jsx:72]: love this, great idea to use the default props

**Review by tay1orjones — APPROVED** [review]
looks great!

**Pull request merged**